src/site/antora/modules/ROOT/pages/manual/plugins.adoc

```diff
@@ -227,15 +227,23 @@ Maven::
   <groupId>org.apache.maven.plugins</groupId>
   <artifactId>maven-compiler-plugin</artifactId>
   <version>${maven-compiler-plugin.version}</version>
+  <configuration>
+    <compilerArgs>
+        <!-- Provide the project coordinates to `GraalVmProcessor`: -->
+        <arg>-Alog4j.graalvm.groupId=${project.groupId}</arg>
+        <arg>-Alog4j.graalvm.artifactId=${project.artifactId}</arg>
+    </compilerArgs>
+  </configuration>
   <executions>
     <execution>
-      <id>generate-log4j-plugin-descriptor</id>
-      <goals>
-        <goal>compile</goal>
-      </goals>
-      <phase>process-classes</phase>
+        <!--
+          ~ Explicitly list the annotation processors for the default compile execution.
+          ~ This is required starting with JDK 23, where annotation processors are not enabled automatically.
+          ~ Explicit configuration also improves build reliability and clarity.
+          ~ For more details, see: https://inside.java/2024/06/18/quality-heads-up/
+          -->
+      <id>default-compile</id>
       <configuration>
-        <proc>only</proc>
```

Comment:
Why did we remove this configuration?

Reply:
I switched from a custom processor only execution to `default-compile`, which should also compile the code.